server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -361,6 +375,136 @@ public void test_jwt_authentication_claim_mismatch_throws_error() throws Excepti
             .hasMessage("jwt authentication failed for user John. Reason: The Claim 'iss' value doesn't match the required issuer.");
     }
 
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_default_is_used() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        var settings = Settings.builder()
+            // Matches JWT_TOKEN payload.
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_ISS_SETTING.getKey(), "https://console.cratedb-dev.cloud/api/v2/meta/jwk/")
+            // Matches JWT_TOKEN payload.
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_AUD_SETTING.getKey(), "test_cluster_id")
+            .build();
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            settings,
+            jwkProviderFunction(null),
+            () -> null // ClusterId supplier is not used, aud is taken from defaults
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        Role authenticatedRole = jwtAuth.authenticate(credentials, null);
+        assertThat(authenticatedRole.name()).isEqualTo(tokenUsername);
+    }
+
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_aud_is_not_defined_in_default() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        var settings = Settings.builder()
+            // issuer Matches JWT_TOKEN payload.
+            // No default value for aud, must use clusterId.
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_ISS_SETTING.getKey(), "https://console.cratedb-dev.cloud/api/v2/meta/jwk/")
+            .build();
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            settings,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        Role authenticatedRole = jwtAuth.authenticate(credentials, null);
+        assertThat(authenticatedRole.name()).isEqualTo(tokenUsername);
```

Comment:
Should the `aud` be verified only as the test name indicates? Talking about `aud` but asset on the username is really confusing.

Reply:
Not sure how we can explicitly verify `aud`. It's done implicitly in authenticate when value hardcoded token compared with what we provide in test setup in Supplier.

Success of `authenticate` implies that `aud` check passed (along with others) despite on being absent in user properties and not being specified in defaults.

Reply:
I removed this test - token's aud is checked with clusterId, for that we already have existing `test_jwt_authentication_default_aud_different_as_token`